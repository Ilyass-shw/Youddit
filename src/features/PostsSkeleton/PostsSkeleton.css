.post-data-container {
	display: flex;
	flex-direction: column;
}
.subreddit span:nth-child(2) {
	margin-left: 4px;
}
.skeleton-content-image{
	width: auto;
}