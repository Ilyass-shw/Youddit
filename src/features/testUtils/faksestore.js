import { configureStore } from "@reduxjs/toolkit";
import { createSlice } from "@reduxjs/toolkit";

const initialState = {
	firstFilter: "relevance",
	lastFilter: "comments",
	filter: "relevance",
};
const fakeSlice = createSlice({
	name: "posts",
	initialState,
	reducers: {},
});

export default configureStore({
	reducer: {
		posts: fakeSlice.reducer,
	},
});

export const makeCustomTestStore = (name, initialState, reducers = {}) => {
	const fakeSlice = createSlice({
		name,
		initialState,
		reducers,
	});

	return configureStore({
		reducer: {
			posts: fakeSlice.reducer,
		},
	});
};
