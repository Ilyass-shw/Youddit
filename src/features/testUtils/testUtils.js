import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import store, { makeCustomTestStore } from "./faksestore";

const TestProvider = ({ store, children }) => <Provider store={store}>{children}</Provider>;

export function render(ui, { store, ...otherOpts }) {
	return rtlRender(<TestProvider store={store}>{ui}</TestProvider>, otherOpts);
}

export function makeTestStore(name, initialState, reducers) {
	if (name && initialState) {
		const Store = makeCustomTestStore(name, initialState, reducers);
		const origDispatch = Store.dispatch;
		Store.dispatch = jest.fn(origDispatch);
		return Store;
	} else {
		const origDispatch = store.dispatch;
		store.dispatch = jest.fn(origDispatch);
		return store;
	}
}

// re-export everything
export * from "@testing-library/react";
