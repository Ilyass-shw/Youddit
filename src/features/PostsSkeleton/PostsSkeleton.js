import React from "react";
import "./PostsSkeleton.css";
import Skeleton, { SkeletonTheme } from "react-loading-skeleton";

const PostsSkeleton = () => {
	return (
		<div>
			<SkeletonTheme color="#262626" highlightColor="#444">
				<div className="post-block skeleton-post-block">
					<div className="content-image skeleton-content-image">
						<Skeleton width="240px" height="10rem" />
					</div>

					<div className="post-data skeleton-post-data-mobile">
						<div className="post-data-container skeleton-post-data-container-mobile">
							<Skeleton width="180px" height="20px" />
							<Skeleton width="140px" height="13px" />
						</div>

						<div className="subreddit skeleton-subreddit-mobile">
							<Skeleton className="subreddit-icon" />
							<Skeleton width="80px" height="13px" />
						</div>
					</div>
				</div>
			</SkeletonTheme>
		</div>
	);
};

export default PostsSkeleton;
