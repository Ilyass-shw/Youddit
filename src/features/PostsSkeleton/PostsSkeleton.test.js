import React from "react";
import PostsSkeleton from "./PostsSkeleton";
import { render, makeTestStore } from "../testUtils/testUtils";

let store;
describe("PostsSkeleton", () => {
	beforeEach(() => {
		store = makeTestStore();
	});

	it("should render without crashing", () => {
		render(<PostsSkeleton />, { store });
	});

	it("should match the snapshot", () => {
		const { container } = render(<PostsSkeleton />, { store });
		expect(container).toMatchSnapshot();
	});
});
