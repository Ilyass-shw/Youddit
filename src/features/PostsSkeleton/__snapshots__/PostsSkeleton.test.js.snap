// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsSkeleton should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="css-18kc0t3-SkeletonTheme"
    >
      <div
        class="post-block skeleton-post-block"
      >
        <div
          class="content-image skeleton-content-image"
        >
          <span>
            <span
              class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 240px; height: 10rem;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="post-data skeleton-post-data-mobile"
        >
          <div
            class="post-data-container skeleton-post-data-container-mobile"
          >
            <span>
              <span
                class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style="width: 180px; height: 20px;"
              >
                ‌
              </span>
            </span>
            <span>
              <span
                class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style="width: 140px; height: 13px;"
              >
                ‌
              </span>
            </span>
          </div>
          <div
            class="subreddit skeleton-subreddit-mobile"
          >
            <span>
              <span
                class="react-loading-skeleton subreddit-icon css-1eg0sfo-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
            <span>
              <span
                class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style="width: 80px; height: 13px;"
              >
                ‌
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
